app/src/main/java/com/rajatgoyal/puzzle15/ui/MainActivity.java

```diff
@@ -219,14 +227,51 @@ public void setLatestHighScore(HighScore latestHighScore) {
     }
 
     public void init() {
+        SharedPref.init(this);
+
+        resumeBtn = findViewById(R.id.resumeGame);
+        resumeBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent intent = new Intent(MainActivity.this, GameActivity.class);
```

Comment:
I see similar intents created many times, with same extra parameters. Probably create a function for it which returns an intent so if there are some other parameters which are to be added then also it would work fine